src/client/datascience/kernel-launcher/localKernelSpecFinderBase.ts

```diff
@@ -153,8 +154,21 @@ export abstract class LocalKernelSpecFinderBase {
         kernelJson.name = interpreter ? getInterpreterKernelSpecName(interpreter) : kernelJson.name;
 
         // Update the display name too if we have an interpreter.
+        const isDefaultPythonName = kernelJson.display_name.toLowerCase().match(isDefaultPythonKernelSpecSpecName);
+        if (!isDefaultPythonName && kernelJson.language === PYTHON_LANGUAGE && kernelJson.argv.length > 2) {
+            // Default kernel spec argv for Python kernels is `"python","-m","ipykernel_launcher","-f","{connection_file}"`
+            // Some older versions had `ipykernel` instead of `ipykernel_launcher`
+            // If its different, then use that as an identifier for the kernel name.
+            const argv = kernelJson.argv
+                .slice(1) // ignore python
+                .map((arg) => arg.toLowerCase())
+                .filter((arg) => !['-m', 'ipykernel', 'ipykernel_launcher', '-f', '{connection_file}'].includes(arg));
+            if (argv.length) {
+                kernelJson.name = `${kernelJson.name}.${argv.join('#')}`;
```

Comment:
The name is stored in the metadata, we need to ensure the name is not the same if we have different args.
Will need to do the same if we have kernels with env variables (again, seperate issue)